views/js/controller/runtime/testRunner.js

```diff
@@ -7,6 +7,51 @@ define(['jquery', 'lodash', 'spin', 'serviceApi/ServiceApi', 'serviceApi/UserInf
 		var timeDiffs = [];
 		var waitingTime = 0;
 
+        /**
+         * Provides a versatile progress bar updater
+         * @type {{update: Function, percentageProgression: Function, positionProgression: Function}}
+         */
+        var progressUpdaters = {
+            /**
+             * Updates the progress bar
+             * @param {Object} assessmentTestContext The progression context
+             */
+            update: function(assessmentTestContext) {
+                var progressIndicator = assessmentTestContext.progressIndicator || 'percentage';
+                var progressIndicatorMethod = progressIndicator + 'Progression';
+                var getProgression = this[progressIndicatorMethod] || this.percentageProgression;
+                var progression = getProgression(assessmentTestContext);
+
+                $('#qti-progress-label').text(progression.label);
+                $('#qti-progressbar').progressbar('value', progression.ratio);
+            },
+
+            /**
+             * Updates the progress bar displaying the percentage
+             * @param {Object} assessmentTestContext The progression context
+             */
+            percentageProgression: function(assessmentTestContext) {
+                var ratio = Math.floor(assessmentTestContext.numberCompleted / assessmentTestContext.numberItems * 100);
+                return {
+                    ratio : ratio,
+                    label : __('Test completed at %d%%').replace('%d', ratio).replace('%%', '%')
+                };
+            },
+
+            /**
+             * Updates the progress bar displaying the position
+             * @param {Object} assessmentTestContext The progression context
+             */
+            positionProgression: function(assessmentTestContext) {
+                var total = assessmentTestContext.numberItems;
+                var position = assessmentTestContext.itemPosition + 1;
+                return {
+                    ratio : Math.floor(position / total * 100),
+                    label : __('Item %X on %Y').replace('%X', position).replace('%Y', total)
```

Comment:
the function  `__` accept basic formatting :  `__('Item %d on %d', position, total)` (syntax to be verified)